packages/flutter_tools/lib/src/mdns_discovery.dart

```diff
@@ -57,6 +57,7 @@ class MDnsObservatoryDiscovery {
       final List<PtrResourceRecord> pointerRecords = await client
           .lookup<PtrResourceRecord>(
             ResourceRecordQuery.serverPointer(dartObservatoryName),
+            timeout: const Duration(seconds: 30),
```

Comment:
Can we send a PR to `package:multicast_dns` to allow timeout to be null to wait forever?

Reply:
Yes. That would have the effect of delegating to the OS-level timeout. OTOH, it may be desirable to have control over the timeout rather than delegating to the OS, and getting an empty-list return rather than handling the async SocketException. wdyt?